templates/sources.lists.j2

```diff
@@ -0,0 +1,75 @@
+# {{ ansible_managed }}
+{%- macro write_entry(suite, release_path) %}
+{{- fmtstr|format('deb', mirror, suite, release, release_path, components) }}
+{%- if gen_sources %}
+	{{- fmtstr|format('deb-src', mirror, suite, release, release_path, components) }}
+{%- endif %}
+{%- endmacro %}
+
+{%- macro write_additional_entries() %}
+{% if additional_source_strings|length >= 1 %}
+
+## Additional repositories
+{% for repo in additional_source_strings %}
+{{ repo }}
+{% endfor %}
+{% endif %}
+{%- endmacro %}
+
+
+{%- if ansible_distribution == 'Debian' %}
+
+{% set mirror = apt_sources_debian.mirror|default(apt_sources_debian_mirror) %}
+{% set release = apt_sources_debian.release|default(ansible_distribution_release) %}
+{% set components = apt_sources_debian.components|default(apt_sources_debian_components)|join(' ') %}
+{% set gen_sources = apt_sources_debian.enable_sources|default(apt_sources_debian_enable_sources) %}
+{% set additional_source_strings = apt_sources_debian.additional_source_strings|default(apt_sources_debian_additional_source_strings) %}
+{%- if gen_sources %}
+	{%- set fmtstr='%-7s %s%s %s%s %s' %}
+{%- else %}
+	{%- set fmtstr='%-3s %s%s %s%s %s' %}
+{%- endif %}
+
+## Main repositories
+{{ write_entry('debian/', '') }}
+{% if apt_sources_debian.security_repo|default(apt_sources_debian_security_repo) %}
+	{{- write_entry('debian-security/', '/updates') }}
+{% endif %}
+{% if apt_sources_debian.update_repo|default(apt_sources_debian_update_repo) %}
+	{{- write_entry('debian/', '-updates') }}
+{% endif %}
+{% if apt_sources_debian.backports|default(apt_sources_debian_backports) %}
+	{{- write_entry('debian/', '-backports') }}
+{% endif %}
+{{ write_additional_entries() }}
```

Comment:
You could pull this out of the `if`, because it is the same for both distributions.

Reply:
I found it more clear to have both sections separated and to not mix them. Specially since both distributions tend to change them independently.